src/app/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,36 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+import { TodoItem } from './useItemsStore';
+
+export function useLocalStorage(key: string, defaultItems: TodoItem[]) {
+	const [value, setValue] = useState(() => {
+		try {
+			const storedValue = localStorage.getItem(key);
+
+			return storedValue
+				? JSON.parse(storedValue)
+				: JSON.parse(
+						localStorage.getItem('items') ??
+							JSON.stringify(defaultItems),
+					);
+		} catch (error) {
+			console.error(error);
+
+			return JSON.parse(
+				localStorage.getItem('items') ?? JSON.stringify(defaultItems),
+			);
+		}
+	});
+
+	useEffect(() => {
+		try {
+			localStorage.setItem(key, JSON.stringify(value));
+		} catch (error) {
+			console.error(error);
+		}
+	}, [key, value]);
+
+	return [value, setValue];
```

Comment:
I would probably choose a little different approach here that I believe is more straightforward:
```ts
function getInitialValue(key: string, defaultItems: TodoItem[]) {
	try {
		const item = localStorage.getItem(key);

		return item ? JSON.parse(item) : defaultItems;
	} catch {
		return [];
	}
}

export function useLocalStorage(key: string, defaultItems: TodoItem[]) {
	const [value, setValue] = useState<TodoItem[]>(
		getInitialValue(key, defaultItems),
	);

	const storeValue = (items: TodoItem[]) => {
		setValue(items);
		localStorage.setItem(key, JSON.stringify(items));
	};

	return [value, storeValue] as const;
}

```